### PR TITLE
Validate for Kubernetes Provider

### LINKED ISSUE #782: Implement Validate method for Kubernetes Provider
Currently, kubernetes provider doesn't have a Validate method implemented.

This would benefit users to know if a given kubernetes configuration is valid or not.

### PR DESCRIPTION
**Pull request by burak-yuksel**
Creating Validate method for Kubernetes provider
Fixes #782

### CODE REVIEW

**Comment by gusfcarvalho** [comment]
Closing and reopening to trigger actions.

File: pkg/provider/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -231,5 +240,25 @@ func (k *BaseClient) fetchSecretKey(ctx context.Context, key esmeta.SecretKeySel
 }
 
 func (k *ProviderKubernetes) Validate() error {
+	ctx := context.Background()
+
+	authReview, err := k.ReviewClient.Create(ctx, &authv1.SelfSubjectAccessReview{
```
Comment (moolen) [inline pkg/provider/kubernetes/kubernetes.go:245]: Can we add a small note to the docs that the `SelfSubjectAccessReview` permission is required for the service account?

Comment (burak-yuksel) [reply pkg/provider/kubernetes/kubernetes.go:243]: @moolen thank you for your feedback. I have added a small note to the documentation.

**Review by moolen — APPROVED** [review]
Sorry for the delay, i've finally found some time to put my hands on it. Works like a charm! :100:

**Comment by gusfcarvalho** [comment]
Thanks for your awesome contribution @burak-yuksel!

**Pull request merged**